sympy/vector/lamecoeff.py

```diff
@@ -0,0 +1,58 @@
+from sympy.core.basic import Basic
+from sympy import sqrt, sin, cos
+from sympy.core.function import Derivative
+
+
+class CoeffProvider(Basic):
```

Comment:
What's the purpose of this class? What mathematical concept does this class represent? (if you inherit `Basic`, you're supposed to be mapping a mathematical concept).

Do you want to have a objects to represent the differential structure of the coordinate system? (that is, an object shared by all coordinate systems having the same differential behaviour?)